### PR TITLE
pass the value to the input instead to downshift

### PR DESCRIPTION
**Pull request by boyskila**

### CODE REVIEW

**Comment by juliusv** [comment]
@boyskila Could you rebase this on master? (https://github.com/prometheus/prometheus/pull/6206 got merged in the meantime)

**Comment by boyskila** [comment]
> @boyskila Could you rebase this on master? (#6206 got merged in the meantime)

Sure, but a bit later. I don't have access to my local machine now

**Comment by juliusv** [comment]
:+1: will merge when CI passes

**Comment by juliusv** [comment]
@boyskila React tests seem to need adjustments (see Travis log).

File: web/ui/react-app/src/ExpressionInput.test.tsx [hunk 1]
```diff
@@ -28,8 +28,8 @@ describe('ExpressionInput', () => {
   const expressionInput = mount(<ExpressionInput {...expressionInputProps} />);
 
   it('renders a downshift component', () => {
-    const downshift = expressionInput.find(Downshift);
-    expect(downshift.prop('inputValue')).toEqual('node_cpu');
+    const downshift = expressionInput.find(Input);
+    expect(downshift.prop('value')).toEqual('node_cpu');
```
Comment (juliusv) [inline web/ui/react-app/src/ExpressionInput.test.tsx]: While this makes the test pass, it no longer fits to the test goal of `it('renders a downshift component', ...)`.

Comment (juliusv) [reply web/ui/react-app/src/ExpressionInput.test.tsx]: (I'm fine changing it in either direction, whichever you think makes more sense)

Comment (boyskila) [reply web/ui/react-app/src/ExpressionInput.test.tsx]: Yes, will change it. One thing I notice here is the shared instance of ExpressionInput between tests. I think each test should operate over a fresh instance of the Input and tests not to depend on each other.

Comment (juliusv) [reply web/ui/react-app/src/ExpressionInput.test.tsx]: Yeah, makes sense to me, even though in this instance it's probably not super important. /cc @hooten

Comment (hooten) [reply web/ui/react-app/src/ExpressionInput.test.tsx]: That sounds good to me!

**Comment by hooten** [comment]
Question (for perhaps another pr; asking here bc `value` is in play):

 It seems like `PanelList` is actually managing the state of `value` [here](https://github.com/prometheus/prometheus/blob/master/web/ui/react-app/src/pages/PanelList.tsx#L128) and [here](https://github.com/prometheus/prometheus/blob/master/web/ui/react-app/src/pages/PanelList.tsx#L36). But there is still a React [Ref](https://github.com/prometheus/prometheus/blob/master/web/ui/react-app/src/ExpressionInput.tsx#L34) that also seems to play a role. Why do we need that ref?

**Comment by boyskila** [comment]
@hooten to control the height of the input. If you press shift + enter the input(actually is a text area) will grow in height. In this case ```offsetHeight, clientHeight, scrollHeight``` can be taken from event.target but there are few other cases where no event exist..so ref to the rescue :)

**Comment by juliusv** [comment]
@boyskila Sorry, could you do one more conflict resolution? I promise I'll remember to merge it then :)

**Comment by boyskila** [comment]
@juliusv branch is rebased with master and no changes are pushed since last commit

**Comment by juliusv** [comment]
:+1: thanks!

**Pull request merged**